### PR TITLE
Parameter reader updated. This should fix PR#1054s numerical issues

### PR DESCRIPTION
**Pull request by vikaskurapati**

### CODE REVIEW

File: src/DynamicRupture/FrictionLaws/LinearSlipWeakening.h [hunk 1]
```diff
@@ -2,6 +2,7 @@
 #define SEISSOL_LINEARSLIPWEAKENING_H
 
 #include "BaseFrictionLaw.h"
+#include "SeisSol.h"
```
Comment (sebwolf-de) [inline src/DynamicRupture/FrictionLaws/LinearSlipWeakening.h:5]: Can be removed IMHO.

File: src/Equations/damaged-elastic/Kernels/DeviceAux/KernelsAux.h [hunk 2]
```diff
@@ -0,0 +1,90 @@
+#pragma once
```
Comment (sebwolf-de) [inline src/Equations/damaged-elastic/Kernels/DeviceAux/KernelsAux.h:1]: Is this supposed to be a hard copy or could it be symlinked?

File: src/Equations/damaged-elastic/Kernels/DeviceAux/cuda/KernelsAux.cu [hunk 3]
```diff
@@ -0,0 +1,338 @@
+#include <Kernels/precision.hpp>
```
Comment (sebwolf-de) [inline src/Equations/damaged-elastic/Kernels/DeviceAux/cuda/KernelsAux.cu:1]: Is this supposed to be a hard copy or could it be symlinked?

File: src/Equations/damaged-elastic/Kernels/DeviceAux/hip/KernelsAux.cpp [hunk 4]
```diff
@@ -0,0 +1,358 @@
+#include <Kernels/precision.hpp>
```
Comment (sebwolf-de) [inline src/Equations/damaged-elastic/Kernels/DeviceAux/hip/KernelsAux.cpp:1]: Is this supposed to be a hard copy or could it be symlinked?

File: src/Equations/damaged-elastic/Kernels/DeviceAux/sycl/KernelsAux.cpp [hunk 5]
```diff
@@ -0,0 +1,271 @@
+#include <CL/sycl.hpp>
```
Comment (sebwolf-de) [inline src/Equations/damaged-elastic/Kernels/DeviceAux/sycl/KernelsAux.cpp:1]: Is this supposed to be a hard copy or could it be symlinked?

File: src/Equations/damaged-elastic/Kernels/Time.cpp [hunk 6]
```diff
@@ -183,18 +183,21 @@ void seissol::kernels::Time::computeAder(double i_timeStepWidth,
...
+
+#ifdef USE_DAMAGEDELASTIC
+  const real epsInitxx = damagedElasticParameters->epsInitxx; // eps_xx0
+  const real epsInityy = damagedElasticParameters->epsInityy; // eps_yy0
+  const real epsInitzz = damagedElasticParameters->epsInitzz; // eps_zz0
+  const real epsInitxy = damagedElasticParameters->epsInitxy; // eps_xy0
+  const real epsInityz = damagedElasticParameters->epsInityz; // eps_yz0
+  const real epsInitzx = damagedElasticParameters->epsInitzx; // eps_zx0
+
+  real const damagedParameter = data.material.local.Cd; // 1.2e-4*2;
```
Comment (sebwolf-de) [inline src/Equations/damaged-elastic/Kernels/Time.cpp:196]: Remove comments

Comment (sebwolf-de) [inline src/Equations/damaged-elastic/Kernels/Time.cpp:196]: Is it a parameter describing the damage (damageParameter) or is the parameter damaged?

File: src/Initializer/Parameters/ModelParameters.cpp [hunk 7]
```diff
@@ -33,20 +33,21 @@ ITMParameters readITMParameters(ParameterReader* baseReader) {
...
-  const auto aB1 = reader->readWithDefault<real>("ab1", 0);
-  const auto aB2 = reader->readWithDefault<real>("ab2", 0);
-  const auto aB3 = reader->readWithDefault<real>("ab3", 0);
-  const auto scalingvalue = reader->readWithDefault<real>("scalingvalue", 0);
+  const auto betaAlpha = reader->readWithDefault<real>("betaalpha", 0.05);
+  const auto aB0 = reader->readWithDefault<real>("ab0", 7.43e9);
+  const auto aB1 = reader->readWithDefault<real>("ab1", -12.14e9);
+  const auto aB2 = reader->readWithDefault<real>("ab2", 18.93e9);
+  const auto aB3 = reader->readWithDefault<real>("ab3", -5.067e9);
+  const auto scalingValue = reader->readWithDefault("scalingvalue", 1e2);
```
Comment (sebwolf-de) [inline src/Initializer/Parameters/ModelParameters.cpp:50]: I'd still vouch for 0 as default values.

File: src/Kernels/DynamicRupture.h [hunk 8]
```diff
@@ -66,6 +66,7 @@ class seissol::kernels::DynamicRupture {
   private:
     dynamicRupture::kernel::evaluateAndRotateQAtInterpolationPoints m_krnlPrototype;
     kernels::Time m_timeKernel;
+    seissol::initializer::parameters::DamagedElasticParameters* damagedElasticParameters;
```
Comment (sebwolf-de) [inline src/Kernels/DynamicRupture.h:69]: I'd add the `m_` prefix here to be consistent with the rest of the code in this file ....

File: src/Kernels/DynamicRupture.h [hunk 9]
```diff
@@ -79,7 +80,7 @@ class seissol::kernels::DynamicRupture {
     std::array<std::shared_ptr<basisFunction::SampledTimeBasisFunctions<real>>, CONVERGENCE_ORDER> timeBasisFunctions;
 #endif
 
-  DynamicRupture( const seissol::initializer::parameters::DamagedElasticParameters* damageParameters ) : m_damageParameters(damageParameters) {}
+  DynamicRupture(seissol::initializer::parameters::DamagedElasticParameters* mDamagedElasticParameters);
```
Comment (sebwolf-de) [inline src/Kernels/DynamicRupture.h:83]: .... and remove the `m` here.

File: src/Kernels/Local.h [hunk 10]
```diff
@@ -80,17 +80,17 @@ class seissol::kernels::Local : public LocalBase {
...
     void flopsIntegral(FaceType const i_faceTypes[4],
                        unsigned int &o_nonZeroFlops,
                        unsigned int &o_hardwareFlops );
                         
     unsigned bytesIntegral();
 
-    void setDamageParameters( seissol::initializer::parameters::DamagedElasticParameters* iDamagedElasticParameters ) {
-      mDamagedElasticParameters = iDamagedElasticParameters;
-    }
+    void const setDamagedElasticParameters(seissol::initializer::parameters::DamagedElasticParameters* mDamagedElasticParameters){
```
Comment (sebwolf-de) [inline src/Kernels/Local.h:91]: Why `void const`?

File: src/Kernels/Local.h [hunk 11]
```diff
@@ -80,17 +80,17 @@ class seissol::kernels::Local : public LocalBase {
...
+      damagedElasticParameters = mDamagedElasticParameters;
```
Comment (sebwolf-de) [inline src/Kernels/Local.h:92]: Also here, I suppose to move the `m` prefix, such that the member of the class has the prefix and the parameter does not have the prefix.

File: src/Kernels/Receiver.cpp [hunk 12]
```diff
@@ -2,7 +2,8 @@
  * @file
```
Comment (sebwolf-de) [inline src/Kernels/Receiver.cpp:2]: Did you apply clang-format to this file? Then please also add it to the allowlist in `.ci/format.sh`

File: src/Kernels/Receiver.h [hunk 13]
```diff
@@ -88,7 +88,7 @@ namespace seissol {
           seissolInstance(seissolInstance)
       {}
 
-      ReceiverCluster(  GlobalData const*             global,
+ReceiverCluster(  GlobalData const*             global,
```
Comment (sebwolf-de) [inline src/Kernels/Receiver.h:91]: Why did you change this? Please keep the diff of the PR as small as possible. Otherwise the git history of this file will be a nightmare to read. In a year, nobody will understand, why this line was changed.

File: src/Kernels/Time.h [hunk 14]
```diff
@@ -159,10 +158,12 @@ class seissol::kernels::Time : public TimeBase {
...
 
   unsigned int* getDerivativesOffsets();
-
-  void setDamagedElasticParameters(seissol::initializer::parameters::DamagedElasticParameters* idamagedElasticParameters) {
-      damagedElasticParameters = idamagedElasticParameters;
-    }
+  /**
+   * sets the damaged elastic parameters pointer
+  */
+  void const setDamagedElasticParameters(seissol::initializer::parameters::DamagedElasticParameters* mDamagedElasticParameters){
```
Comment (sebwolf-de) [inline src/Kernels/Time.h:164]: As above: Remove `const`, swap `m`

File: src/Solver/time_stepping/TimeCluster.cpp [hunk 15]
```diff
@@ -400,20 +400,19 @@ void seissol::time_stepping::TimeCluster::computeLocalIntegration(seissol::initi
...
+    //TODO(NONLINEAR) Where do these numbers come from?
+    const auto& damagedElasticParameters = seissolInstance.getSeisSolParameters().model.damagedElasticParameters;
+    
+    const real epsInitxx = damagedElasticParameters.epsInitxx; // eps_xx0
+    const real epsInityy = damagedElasticParameters.epsInityy; // eps_yy0
+    const real epsInitzz = damagedElasticParameters.epsInitzz; // eps_zz0
+    const real epsInitxy = damagedElasticParameters.epsInitxy; // eps_xy0
+    const real epsInityz = damagedElasticParameters.epsInityz; // eps_yz0
+    const real epsInitzx = damagedElasticParameters.epsInitzx; // eps_zx0
+    real const damageParameter = data.material.local.Cd; // 1.2e-4*2;
```
Comment (sebwolf-de) [inline src/Solver/time_stepping/TimeCluster.cpp:412]: Remove comments.

File: src/Solver/time_stepping/TimeCluster.cpp [hunk 16]
```diff
@@ -400,20 +400,19 @@ void seissol::time_stepping::TimeCluster::computeLocalIntegration(seissol::initi
...
+    const real scalingValue = damagedElasticParameters.scalingValue;
+    real const breakCoefficient = scalingValue*damageParameter;
+    real const betaAlpha = damagedElasticParameters.betaAlpha;
```
Comment (sebwolf-de) [inline src/Solver/time_stepping/TimeCluster.cpp:415]: Be consistent with `real const` and `const real`.

File: src/Solver/time_stepping/TimeCluster.cpp [hunk 17]
```diff
@@ -1080,19 +1079,15 @@ void seissol::time_stepping::TimeCluster::updateMaterialLocal(seissol::initializ
...
-    const real aB1 = damagedElasticParameters.aB1;
-    const real aB2 = damagedElasticParameters.aB2;
-    const real aB3 = damagedElasticParameters.aB3;
+      // TODO(NONLINEAR) Write generalized reader for these parameters
+      const real epsInitxx = damagedElasticParameters.epsInitxx; // eps_xx0
+      const real epsInityy = damagedElasticParameters.epsInityy; // eps_yy0
+      const real epsInitzz = damagedElasticParameters.epsInitzz; // eps_zz0
+      const real epsInitxy = damagedElasticParameters.epsInitxy; // eps_xy0
+      const real epsInityz = damagedElasticParameters.epsInityz; // eps_yz0
+      const real epsInitzx = damagedElasticParameters.epsInitzx; // eps_zx0
```
Comment (sebwolf-de) [inline src/Solver/time_stepping/TimeCluster.cpp:1090]: Remove comments.

File: src/Solver/time_stepping/TimeCluster.cpp [hunk 18]
```diff
@@ -1112,9 +1107,14 @@ void seissol::time_stepping::TimeCluster::updateMaterialLocal(seissol::initializ
     const real alphaAve = Q_aveData[9];
     const real breakAve = Q_aveData[10];
 
-    real lambda0 = materialData[l_cell].local.lambda0;
-    real mu0 = materialData[l_cell].local.mu0;
+      real lambda0 = materialData[l_cell].local.lambda0;
+      real mu0 = materialData[l_cell].local.mu0;
 
+      // TODO(NONLINEAR) Write generalized reader for these parameters
```
Comment (sebwolf-de) [inline src/Solver/time_stepping/TimeCluster.cpp:1113]: Remove comment

File: src/Solver/time_stepping/TimeCluster.h [hunk 19]
```diff
@@ -381,29 +381,27 @@ class seissol::time_stepping::TimeCluster : public seissol::time_stepping::Abstr
...
 
             // TODO(NONLINEAR) Write unified reader for these parameters
+    const auto& damagedElasticParameters = seissolInstance.getSeisSolParameters().model.damagedElasticParameters;
+    
+    const real epsInitxx = damagedElasticParameters.epsInitxx; // eps_xx0
+    const real epsInityy = damagedElasticParameters.epsInityy; // eps_yy0
+    const real epsInitzz = damagedElasticParameters.epsInitzz; // eps_zz0
+    const real epsInitxy = damagedElasticParameters.epsInitxy; // eps_xy0
+    const real epsInityz = damagedElasticParameters.epsInityz; // eps_yz0
+    const real epsInitzx = damagedElasticParameters.epsInitzx; // eps_zx0
```
Comment (sebwolf-de) [inline src/Solver/time_stepping/TimeCluster.h:391]: Remove comment.

File: src/Solver/time_stepping/TimeCluster.h [hunk 20]
```diff
@@ -381,29 +381,27 @@ class seissol::time_stepping::TimeCluster : public seissol::time_stepping::Abstr
...
+
+            real lambda0P = materialData[l_cell].local.lambda0;
+            real mu0P = materialData[l_cell].local.mu0;
+            real rho0P = materialData[l_cell].local.rho;
+
+            real lambda0M = materialData[l_cell].neighbor[side].lambda0;
+            real mu0M = materialData[l_cell].neighbor[side].mu0;
+            real rho0M = materialData[l_cell].neighbor[side].rho;
```
Comment (sebwolf-de) [inline src/Solver/time_stepping/TimeCluster.h:399]: Indentation? Can these variables be const?

**Review by sebwolf-de — CHANGES_REQUESTED** [review]
Hi, the changes look good, most of my comments are about superficial details.

File: src/Equations/damaged-elastic/Kernels/Time.cpp (hunk 6 above)
Comment (vikaskurapati) [reply src/Equations/damaged-elastic/Kernels/Time.cpp:196]: I don't precisely know here. May be a better name would be damageParameter?

File: src/Kernels/Local.h (hunk 10 above)
Comment (vikaskurapati) [reply src/Kernels/Local.h:91]: It must have been left there as a mistake during the merge. Modified

File: src/Kernels/Receiver.cpp (hunk 12 above)
Comment (vikaskurapati) [reply src/Kernels/Receiver.cpp:2]: I belive so, yes. But I added both the .h and .cpp to the format.sh

File: src/Kernels/Receiver.h (hunk 13 above)
Comment (vikaskurapati) [reply src/Kernels/Receiver.h]: It must've been some formatting error. I applied the clang-format again on the header and .cpp file. I hope the diff is not too different now

File: src/Equations/damaged-elastic/Kernels/Time.cpp (hunk 6 above)
Comment (sebwolf-de) [reply src/Equations/damaged-elastic/Kernels/Time.cpp]: I think `damageParameter` is the better choice. It is written this way in other places.

Comment (vikaskurapati) [reply src/Equations/damaged-elastic/Kernels/Time.cpp]: Done. Thank you

**Review by sebwolf-de — COMMENTED** [review]
Hi, code quality improved with your changes. I only have a few requests.

File: src/Solver/time_stepping/TimeCluster.cpp [hunk 21]
```diff
@@ -1112,9 +1107,13 @@ void seissol::time_stepping::TimeCluster::updateMaterialLocal(seissol::initializ
...
 
-    real lambda0 = materialData[l_cell].local.lambda0;
-    real mu0 = materialData[l_cell].local.mu0;
+      real lambda0 = materialData[l_cell].local.lambda0;
+      real mu0 = materialData[l_cell].local.mu0;
 
+      const real aB0 = damagedElasticParameters.aB0;
+      const real aB1 = damagedElasticParameters.aB1;
+      const real aB2 = damagedElasticParameters.aB2;
+      const real aB3 = damagedElasticParameters.aB3;
```
[the remark this answers is not in the document]
Comment (vikaskurapati) [reply src/Solver/time_stepping/TimeCluster.cpp:1116]: No, they are being reassigned in another loop underneath

**Comment by sebwolf-de** [comment]
Only one thing left: either add `src/Kernels/DynamicRupture.cpp` to the allowlist in `.ci/format.sh` or revert the clang-format changes in this file.

File: .ci/format.sh [hunk 22]
```diff
@@ -35,13 +35,14 @@ format() {
...
+        src/Kernels/DynamicRupture.cpp
         src/Kernels/PointSourceCluster.h
         src/Kernels/PointSourceClusterOnHost.h
         src/Kernels/PointSourceClusterOnHost.cpp
         src/Kernels/PointSourceClusterOnDevice.h
         src/Kernels/PointSourceClusterOnDevice.cpp
-	src/Kernels/Receiver.h
-	src/Kernels/Receiver.cpp
+	    src/Kernels/Receiver.h
+	    src/Kernels/Receiver.cpp
```
Comment (sebwolf-de) [inline .ci/format.sh:45]: Please fix indendation here.

Comment (vikaskurapati) [reply .ci/format.sh]: Weirdly, the VS code shows the correct indentation, but it is off on Git Hub. But I fixed it, nevertheless. Thank you

Comment (sebwolf-de) [reply .ci/format.sh]: Maybe a tab/spaces mismatch?

**Review by sebwolf-de — APPROVED** [review]
